poptorch_experimental_addons/_impl/core.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
```

Comment:
function naming and signature suggestions:

`def autograd_proxy(fwd, proxy)`: provide a `proxy` tensor to replace `fwd` during autograd. Pros: i) function name more closely represents what you are doing, ii) short Cons: i) no precedent for calling it this, ii) must explain what it is.

`def surrogate_grad(fwd, surrogate)`: provides a surrogate tensor to compute gradients with instead of `fwd`. Pros: uses terminology adopted by community for solving a similar problem, Cons: API is not the same (surrogate should be the surrogate gradient itself not surrogate fwd.

`def surrogate_fwd(fwd, surrogate)`: provide a surrogate tensor to compute gradients with instead of fwd. Pros: closely represents what you are doing, Cons: atypical usage of `surrogate` term.

`def custom_grad_with_fwd_proxy(fwd, proxy): provide a proxy tensor to compute a custom gradient with instead of `fwd`. Pros) Uses familiar terminology while explaining difference. Cons: verbose.

Reply:
Thanks for this - very good review of the options. I think I like `autograd_proxy` or `custom_grad_with_fwd_proxy` personally. @thecharlieblake / @lukaribar any opinion?